src/webview/pipeline/app/editor.ts

```diff
@@ -0,0 +1,293 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import { createDiv } from './util';
+import { Trigger } from './types';
+import { debounce } from 'debounce';
+
+export interface Widget {
+  getElement(): HTMLElement;
+}
+export class BaseWidget implements Widget {
+  protected element: HTMLElement;
+
+  getElement(): HTMLElement {
+    return this.element;
+  }
+}
+
+export class NavigationItem extends BaseWidget {
```

Comment:
This comment is optional. As a rule, I prefer to keep it one class per file. Multiple classes in a file do not effect performance, that is why I say it is optional. This is a general style that is followed but not a requirement. 

However, I separate them because it is easier to reason over them, find them, and test them.